src/networking.c

```diff
@@ -2138,12 +2136,19 @@ void readQueryFromClient(connection *conn) {
     if (c->reqtype == PROTO_REQ_MULTIBULK && c->multibulklen && c->bulklen != -1
         && c->bulklen >= PROTO_MBULK_BIG_ARG)
     {
-        ssize_t remaining = (size_t)(c->bulklen+2)-sdslen(c->querybuf);
         big_arg = 1;
+        if (c->querybuf) {
+            ssize_t remaining = (size_t) (c->bulklen + 2) - sdslen(c->querybuf);
```

Comment:
```suggestion
            ssize_t remaining = (size_t) (c->bulklen + 2) - (c->querybuf ? sdslen(c->querybuf) : 0);
```
If we just skip it, it will take 2 read(2) to read the big arg.